src/storage/DataAccessorBasedStore.ts

```diff
@@ -382,18 +388,22 @@ export class DataAccessorBasedStore implements ResourceStore {
     // Solid, §5.3: "Servers MUST create intermediate containers and include corresponding containment triples
     // in container representations derived from the URI path component of PUT and PATCH requests."
     // https://solid.github.io/specification/protocol#writing-resources
-    const modified = [];
+    let changes: Record<string, RepresentationMetadata> = {};
     if (!this.identifierStrategy.isRootContainer(identifier) && !exists) {
-      const container = this.identifierStrategy.getParentContainer(identifier);
+      const parent = this.identifierStrategy.getParentContainer(identifier);
       if (!createContainers) {
-        modified.push(container);
+        changes[parent.path] = new RepresentationMetadata(parent).add(SOLID_AS.terms.Activity, AS.Update);
```

Comment:
Since this is something that occurs several times, it might be interesting to have a `createActivityMetadata` helper function here that takes as input an identifier and an activity type and outputs a metadata object.